firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/NotificationFeedbackTrigger.kt

```diff
@@ -0,0 +1,215 @@
+            )
+            channel.description =
+                application.getString(R.string.feedback_notification_channel_description)
+            val notificationManager = application.getSystemService(
+                NotificationManager::class.java
+            )
+            notificationManager.createNotificationChannel(channel)
+        }
+        application.registerActivityLifecycleCallbacks(this)
+    }
+
+    fun enable(currentActivity: Activity? = null) {
+        this.currentActivity = currentActivity
+        isEnabled = true
+        if (currentActivity != null) {
+            showNotification(currentActivity)
+        }
+    }
+
+    fun disable() {
+        isEnabled = false
+        val activity = currentActivity
+        currentActivity = null
+        if (activity != null) {
+            cancelNotification(activity)
+        }
+    }
+
+    private fun showNotification(activity: Activity) {
+        if (ContextCompat.checkSelfPermission(activity, POST_NOTIFICATIONS) == PERMISSION_GRANTED) {
+            val intent = Intent(activity, TakeScreenshotAndTriggerFeedbackActivity::class.java)
+            intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY)
+            val pendingIntent =
+                PendingIntent.getActivity(activity, 0, intent, PendingIntent.FLAG_IMMUTABLE)
+            val builder = NotificationCompat.Builder(activity, FEEBACK_NOTIFICATION_CHANNEL_ID)
+                .setSmallIcon(R.drawable.test_app_icon)
+                .setContentTitle(activity.getText(R.string.feedback_notification_title))
+                .setContentText(activity.getText(R.string.feedback_notification_text))
+                .setPriority(NotificationCompat.PRIORITY_HIGH)
+                .setContentIntent(pendingIntent)
+            val notificationManager = NotificationManagerCompat.from(activity)
+            Log.i(TAG, "Showing notification")
+            notificationManager.notify(FEEDBACK_NOTIFICATION_ID, builder.build())
+        } else {
+            // no permission to post notifications - try to get it
+            if (hasRequestedPermission) {
+                Log.i(
+                    TAG,
+                    "We've already request permission. Not requesting again for the life of the activity."
+                )
+            } else {
+                requestPermission(activity)
+                hasRequestedPermission = true
+            }
+        }
+    }
+
+    private fun requestPermission(activity: Activity) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
+            return // no need
+        }
+        if (activity.shouldShowRequestPermissionRationale(POST_NOTIFICATIONS)) {
+            Log.i(TAG, "Showing customer rationale for requesting permission.")
+            AlertDialog.Builder(activity)
+                .setMessage(
+                    "Using a notification to initiate feedback to the developer. To enable this feature, allow the app post notifications."
```

Comment:
"allow the app **to** post notifications"

Reply:
Done.